F1WM/Repositories/NewsRepository.cs

```diff
@@ -79,7 +79,62 @@ public async Task<NewsDetails> GetNewsDetails(int id)
 			return news;
 		}
 
-		public NewsRepository(F1WMContext context, IMapper mapper)
+        public async Task<IEnumerable<NewsSummary>> GetNewsByTag(int id)
+        {
+            await SetDbEncoding();
+            IEnumerable<News> dbNews;
+            dbNews = await context.News
+                .Include(n => n.Topics)
+                .Where(n => n.Topics.TopicId == id)
```

Comment:
This condition is quite random. After transition to IEnumerable Topics it should use .Any() method on them, so that we will query all topics, not only one per news.